src/target/stm32g0.c

```diff
@@ -679,6 +690,8 @@ static bool stm32g0_cmd_option(target_s *t, int argc, const char **argv)
 	option_register_s options_req[OPT_REG_COUNT] = {};
 
 	if (argc == 2 && strcasecmp(argv[1], "erase") == 0) {
+		if ((t->part_id == STM32C011) || (t->part_id == STM32C031))
```

Comment:
The two pairs of inner parentheses suppress a GCC warning we want on here, please drop them.